packages/eslint-plugin/tests/rules/no-implied-eval.test.ts

```diff
@@ -0,0 +1,640 @@
+      errors: [
+        {
+          messageId: 'noImpliedEvalError',
+          line: 4,
+          column: 12,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 5,
+          column: 13,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 6,
+          column: 14,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 7,
+          column: 12,
+        },
+      ],
+    },
+    {
+      code: `
+const foo: string = 'x + 1';
+
+setTimeout(foo, 0);
+setInterval(foo, 0);
+setImmediate(foo);
+execScript(foo);
+      `,
+      errors: [
+        {
+          messageId: 'noImpliedEvalError',
+          line: 4,
+          column: 12,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 5,
+          column: 13,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 6,
+          column: 14,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 7,
+          column: 12,
+        },
+      ],
+    },
+    {
+      code: `
+const foo = new String('x + 1');
+
+setTimeout(foo, 0);
+setInterval(foo, 0);
+setImmediate(foo);
+execScript(foo);
+      `,
+      errors: [
+        {
+          messageId: 'noImpliedEvalError',
+          line: 4,
+          column: 12,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 5,
+          column: 13,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 6,
+          column: 14,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 7,
+          column: 12,
+        },
+      ],
+    },
+    {
+      code: `
+const foo = 'x + 1';
+
+setTimeout(foo as any, 0);
+setInterval(foo as any, 0);
+setImmediate(foo as any);
+execScript(foo as any);
+      `,
+      errors: [
+        {
+          messageId: 'noImpliedEvalError',
+          line: 4,
+          column: 12,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 5,
+          column: 13,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 6,
+          column: 14,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 7,
+          column: 12,
+        },
+      ],
+    },
+    {
+      code: `
+const fn = (foo: string | any) => {
+  setTimeout(foo, 0);
+  setInterval(foo, 0);
+  setImmediate(foo);
+  execScript(foo);
+}
+      `,
+      errors: [
+        {
+          messageId: 'noImpliedEvalError',
+          line: 3,
+          column: 14,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 4,
+          column: 15,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 5,
+          column: 16,
+        },
+        {
+          messageId: 'noImpliedEvalError',
+          line: 6,
+          column: 14,
+        },
+      ],
+    },
```

Comment:
can you add invalid test case that uses `window.*` and templates \`\`

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals